axlearn/experiments/text/gpt/fuji.py

```diff
@@ -54,17 +105,21 @@ def get_trainer_kwargs(model_size: str, *, vocab_size: int) -> Dict[str, Any]:
                 num_layers=32,
                 hidden_dim=128 * 32,
                 num_heads=32,
+                num_kv_heads=num_kv_heads,
+                rope_theta=rope_theta,
             ),
             learner_kwargs=dict(peak_lr=3e-4, weight_decay=0.1),
-            train_batch_size=4 * 1024 * 1024 // MAX_SEQUENCE_LENGTH,  # 4M tokens.
-            max_step=500_000,  # 2T tokens // 4M tokens/step.
+            max_sequence_length=max_sequence_length,
+            train_batch_size=train_batch_size,
+            max_step=max_step,
             mesh_shape=mesh_shape_from_axes(fsdp=-1),
             mesh_rules=(
                 # tpu-v4. step time: 3.03s.
                 ("tpu-v4-(1024|2048)", mesh_shape_from_axes(data=-1, fsdp=16)),
                 # tpu-v5e. step time: TBD.
                 ("tpu-v5litepod-256", mesh_shape_from_axes(data=-1, fsdp=16)),
                 # H100/A100 80G. Maximum per-node batch size = 64, hence need >= 32 nodes.
+                # p5.48xlarge 8x64. v1 step time: 1.54s.
```

Comment:
Nit, maybe add a comment stating that global batch size = 512 for v3 model and need to adjust the mesh size? 
(1024 GPUs won't work here)